src/getIdentifier.ts

```diff
@@ -0,0 +1,81 @@
+import { VerificationFragment } from "./types/core";
+import { Identity } from "./verifiers/issuerIdentity/dnsTxt";
+import { SignatureVerificationFragment } from "./verifiers/issuerIdentity/did";
+import { DnsVerificationFragment } from "./verifiers/issuerIdentity/dnsDid";
+
+enum IdentityProof {
+  DNS = "OpenAttestationDnsTxtIdentityProof",
+  DNSDID = "OpenAttestationDnsDidIdentityProof",
+  DID = "OpenAttestationDidIdentityProof",
+}
+
+const getDnsIdentifierProof = (fragment: VerificationFragment) => {
+  const type = "DNS";
+  if (Array.isArray(fragment.data)) {
+    return fragment.data.map((issuer: Identity) => ({
+      identifier: issuer.location,
+      type,
+    }));
+  }
+  return {
+    identifier: fragment.data.identifier,
+    type,
+  };
+};
+
+const getDnsDidIdentifierProof = (fragment: VerificationFragment) => {
+  const type = "DNS-DID";
+  if (Array.isArray(fragment.data)) {
+    return fragment.data.map((issuer: DnsVerificationFragment) => ({
+      identifier: issuer.location,
+      type,
+    }));
+  }
+  return {
+    identifier: fragment.data.location,
+    type,
+  };
+};
+
+const getDidIdentifierProof = (fragment: VerificationFragment) => {
+  const type = "DID";
+  if (Array.isArray(fragment.data)) {
+    return fragment.data.map((issuer: SignatureVerificationFragment) => ({
+      identifier: issuer.did,
+      type,
+    }));
+  }
+  return {
+    identifier: fragment.data.did,
+    type,
+  };
+};
+
+export const getIdentityProofFragment = (fragments: VerificationFragment[]) => {
```

Comment:
we already have some types for some fragment, maybe we can enhance and add runtypes as well, so that we are sure about what we handle

Reply:
have added more specific types for each of the functions so that we are sure what we are handling